diff --git a/src/lib/ts/_entry.ts b/src/lib/ts/_entry.ts
index f04710e..a188de6 100644
--- a/src/lib/ts/_entry.ts
+++ b/src/lib/ts/_entry.ts
@@ -77,17 +77,18 @@ register(next => {
 
 		const emit = service.getEmitOutput("/src.ts");
 
-		const diagnostics = new Array<Diagnostic>()
-			.concat(service.getSyntacticDiagnostics("/src.ts"))
-			.concat(service.getSemanticDiagnostics("/src.ts"))
+		const diagnosticMessages = [
+			...service.getSyntacticDiagnostics("/src.ts"),
+			...service.getSemanticDiagnostics("/src.ts")
+		]
 			.map(diagnostic => {
-				const message = flattenDiagnosticMessageText(diagnostic.messageText, "\n");
+				const message = flattenDiagnosticMessageText(diagnostic.messageText, "\n") || "";
 
 				if (diagnostic.file != null) {
 					let file = diagnostic.file.fileName.substring(1);
 					if (file === "src.ts") file = filename;
 
-					if (diagnostic.start == null) return file;
+					if (diagnostic.start == null) return file + ": " + message;
 
 					const pos = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
 					return file + ":" + (pos.line + 1) + ":" + (pos.character + 1) + ": " + message;
@@ -95,8 +96,8 @@ register(next => {
 				else return message;
 			});
 
-		if (diagnostics.length > 0) {
-			throw new SyntaxError(diagnostics.join("\n"));
+		if (diagnosticMessages.length > 0) {
+			throw new SyntaxError(diagnosticMessages.join("\n") as any);
 		}
 
 		const outputs: Record<string, string> = {};
