diff --git a/Gemfile b/Gemfile
index fa87e6a..16929fe 100644
--- a/Gemfile
+++ b/Gemfile
@@ -1,4 +1,4 @@
-source :rubygems
+source 'https://rubygems.org'
 
 gem "paint"
 gem "coderay"
diff --git a/lib/twostroke/runtime/lib/number.rb b/lib/twostroke/runtime/lib/number.rb
index 0df3ecd..206fea6 100644
--- a/lib/twostroke/runtime/lib/number.rb
+++ b/lib/twostroke/runtime/lib/number.rb
@@ -8,7 +8,7 @@ module Twostroke::Runtime
       if this.is_a?(Types::NumberObject)
         base = args[0] ? Types.to_uint32(args[0]) : 10
         Lib.throw_range_error "toString() radix argument must be between 2 and 36" if base < 2 or base > 36
-        if this.number.is_a? Fixnum or this.number.is_a? Bignum
+        if this.number.is_a? Integer
           Types::String.new this.number.to_s base
         else
           Types::String.new this.number.to_s.gsub(/\.0+\z/,"")
diff --git a/lib/twostroke/runtime/types.rb b/lib/twostroke/runtime/types.rb
index d49db8b..20d7448 100644
--- a/lib/twostroke/runtime/types.rb
+++ b/lib/twostroke/runtime/types.rb
@@ -136,7 +136,7 @@ module Twostroke::Runtime::Types
   def self.marshal(ruby_object)
     case ruby_object
     when ::String, Symbol;  String.new ruby_object.to_s
-    when Fixnum, Float;     Number.new ruby_object
+    when Integer, Float;    Number.new ruby_object
     when ::Array;           Array.new ruby_object.map { |el| marshal el }
     when Hash;              o = Object.new
                             ruby_object.each { |k,v| o.put k.to_s, marshal(v) }
diff --git a/lib/twostroke/runtime/vm_frame.rb b/lib/twostroke/runtime/vm_frame.rb
index b22a5ce..6ad1b5f 100644
--- a/lib/twostroke/runtime/vm_frame.rb
+++ b/lib/twostroke/runtime/vm_frame.rb
@@ -96,7 +96,7 @@ module Twostroke::Runtime
     def push(arg)
       if arg.is_a? Symbol
         stack.push scope.get_var(arg)
-      elsif arg.is_a?(Fixnum) || arg.is_a?(Float)
+      elsif arg.is_a?(Integer) || arg.is_a?(Float)
         stack.push Types::Number.new(arg)
       elsif arg.is_a?(String)
         stack.push Types::String.new(arg)
diff --git a/repl.rb b/repl.rb
index c79c979..af9d0a0 100644
--- a/repl.rb
+++ b/repl.rb
@@ -55,12 +55,16 @@ trap "SIGINT" do
   exit!
 end
 
+# Only start REPL if stdin is a TTY and no scripts were executed
+exit if !STDIN.tty? || ARGV.any? { |arg| !arg.start_with?('--') }
+
 loop do
   src = ""
   system "stty raw -echo"
   print Paint[">>> ", :bright]
   loop do
     c = STDIN.getc
+    exit! if c.nil?
     if c.ord == 3
       # ctrl+c
       if src.empty?
